.github/composites/test-coverage-with-artifact/action.yml

```diff
@@ -0,0 +1,44 @@
+# Copyright 2024 Democratized Data Foundation
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+name: 'Test Coverage And Save Artifact'
+
+description: 'Composite action to run the test with coverage and save the report as artifact'
+
+inputs:
+  coverage-artifact-name:
+    description: 'Name of the artifact that will save coverage report'
+    required: true
+    default: 'coverage_default'
+
+  coverage-path:
+    description: 'Path to coverage file'
+    required: false
+    default: 'coverage.txt'
+
+runs:
+  # This is a composite action, setting this is required.
+  using: "composite"
+
+  steps:
+    - name: Run integration tests
+      shell: bash # It's required for run step to specify shell in a composite action.
+      run: make test:coverage
```

Comment:
More like the matrix is running this composite action (bunch of steps like before, now shared instead of duplicating them in every matric job). The composite inherits the env variables from the parent action (so whatever env variables are set when this is called will be applied to them, you can also manually see the env variables for every jobs step when this is called and all the env variables it inherits in the run logs.)

Reply:
Ah got it, that must be the reference `uses: ./.github/composites/test-coverage-with-artifact` in the new jobs then I assume.

Thanks Shahzad :)

Reply:
Yes, correct.